main.ts

```diff
@@ -86,60 +87,7 @@ function createWindow() {
 
 }
 
-// Edited version of a template menu-bar provided by the electron API,
-// refer to https://electronjs.org/docs/api/menu for more information.
-const mainMenuTemplate: Electron.MenuItemConstructorOptions[] = [
-  {
-    label: 'File',
-    submenu: [
-      {
-        label: 'Quit CATcher', accelerator: 'CmdOrCtrl+Q', click() { app.quit(); }
-      }
-    ]
-  },
-  {
-    label: 'Edit',
-    submenu: [
-      { role: 'undo' },
-      { role: 'redo' },
-      { type: 'separator' },
-      { role: 'selectAll' },
-      { role: 'cut' },
-      { role: 'copy' },
-      { role: 'paste' },
-      { role: 'delete' },
-    ]
-  },
-  {
-    label: 'View',
-    submenu: [
-      { role: 'resetZoom' },
-      { role: 'zoomIn' },
-      { role: 'zoomOut' },
-      { type: 'separator' },
-      { role: 'togglefullscreen' }
-    ]
-  },
-  // ,
-  // {
-  //   role: 'help',
-  //   submenu: [
-  //     {
-  //       label: 'User Guide',
-  //       click () { require('electron').shell.openExternal('https://catcher-org.github.io/'); }
-  //     }
-  //   ]
-  // }
-];
-
-if (isDevMode) {
-  let viewSubMenu: MenuItemConstructorOptions[];
-  viewSubMenu = mainMenuTemplate[2].submenu as MenuItemConstructorOptions[];
-  viewSubMenu.push(
-    { type: 'separator' },
-    { role: 'toggleDevTools'}
-  );
-}
+const mainMenuTemplate: MenuItemConstructorOptions[] = createMenuOptions(isDevMode);
```

Comment:
Let's move this declaration into the `ready` event's callback function (line 98).

Reply:
I've moved it into the callback function!